### PR TITLE
win32_deps_build.sh: change Boost URL

### PR DESCRIPTION
**Pull request by petrutlucian94**
The Boost download URL is broken again [1], we'll switch to a working mirror.

Fixes: https://tracker.ceph.com/issues/63952

[1] https://github.com/boostorg/boost/issues/842

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: win32_deps_build.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ sslDir="${depsToolsetDir}/openssl"
 sslSrcDir="${depsSrcDir}/openssl"
 
 # For now, we'll keep the version number within the file path when not using git.
-boostUrl="https://boostorg.jfrog.io/artifactory/main/release/1.82.0/source/boost_1_82_0.tar.gz"
+boostUrl="https://archives.boost.io//release/1.82.0/source/boost_1_82_0.tar.gz"
```
Comment (idryomov) [inline win32_deps_build.sh:19]: double slash
```suggestion
boostUrl="https://archives.boost.io/release/1.82.0/source/boost_1_82_0.tar.gz"
```

Comment (petrutlucian94) [reply win32_deps_build.sh:19]: Whops, thanks.

**Comment by cbodley** [comment]
thanks @petrutlucian94. if the windows build can't use https://download.ceph.com/qa/boost_1_82_0.tar.bz2 because of the .bz2, we might upload the .gz there as well to save on some lab bandwidth

**Comment by petrutlucian94** [comment]
> thanks @petrutlucian94. if the windows build can't use https://download.ceph.com/qa/boost_1_82_0.tar.bz2 because of the .bz2, we might upload the .gz there as well to save on some lab bandwidth

Makes sense, here's a PR that switches to the ceph hosted source: https://github.com/ceph/ceph/pull/55100. bz2 should be ok, I've added bzip2 to the list of requirements to make sure that it's available, especially with containerized builds.

**Pull request merged**